php/Terminus/Models/Environment.php

```diff
@@ -110,6 +110,27 @@ public function backupUrl($bucket, $element) {
   }
 
   /**
+   * Changes connection mode
+   *
+   * @param [string] $value Connection mode, "git" or "sftp"
+   * @return [Workflow] $workflow
+   */
+  public function changeConnectionMode($value) {
+    switch ($value) {
+      case 'git':
+        $workflow_name = 'enable_git_mode';
+        break;
+      case 'sftp':
+        $workflow_name = 'enable_on_server_development';
+        break;
+    }
+
+    $params   = array('environment' => $this->get('id'));
```

Comment:
We can check beforehand, but then:
Best case: 1 check query
Worst case: 1 check query + 1 workflow + n checks to workflow status
As it is now:
Best case: 1 workflow
Worst case: 1 workflow + n checks to workflow status
So doing this is essentially the same in the best-case scenario, but worse in the worst case.